api/apps/api/src/modules/projects/projects-crud.service.ts

```diff
@@ -199,4 +204,57 @@ export class ProjectsCrudService extends AppBaseService<
       return this.planningUnitsService.create(createModel);
     }
   }
+
+  async validateBeforeCreate(
```

Comment:
if we want to avoid mutating in place, we can use `setDataCreate()` and `setDataUpdate()`